docs/java-developer-docs

```diff
@@ -0,0 +1,171 @@
+
+    private String title;
+
+    @ManyToMany
+    @JoinTable(name = "book_author",
+                joinColumns = { @JoinColumn(name = "fk_book") },
+                inverseJoinColumns = { @JoinColumn(name = "fk_author") })
+    private List authors = new ArrayList();
+
+    ...
+}
+
+public interface BookRepository extends CrudRepository<Book, Long> {
+
+    Book findByTitle(String title);
+}
+
+RunWith(SpringRunner.class)
+@SpringBootTest
+public class GettingStartedApplicationTests {
+
+    Logger log = Logger.getLogger(this.getClass().getName());
+
+    @Autowired
+    private BookRepository bookRepository;
+
+
+    @Test
+    @Transactional
+    public void testByTitle() {
+        log.info("... testByTitle ...");
+
+        Book b = bookRepository.findByTitle("Hibernate Tips");
+        Assert.assertEquals(new Long(1), b.getId());
+    }
+
+    @Test
+    @Transactional
+    public void testPersistBook() {
+        log.info("... testPersistBook ...");
+
+        Book b = new Book();
+        b.setTitle("Hibernate Tips - More than 70 solutions to common Hibernate problems");
+
+        bookRepository.save(b);
+    }
+}
+
+
+```
+
+This is limitation and would need fix in in spring-boot itself. Spring-Boot entity framework works on entity<-> repo mapping. This would need fix in spring
+Hence we cannot have movies, genre,actors in the same "movies" collection.
+spring-data-cosmosdb expects to have 3 separate collections as "movies", "genre", "actor"
+However this is cost implication as "genre" collection having just 30 elements needs a separate collection and is 3 times the RU querying for 3 different collection.
+
+Java spring-data-cosmosdb needs patch from Spring framework itself.
+
+
+#### Support for single document read with partition key​ [https://csefy19.visualstudio.com/CSE%20Feedback/_workitems/edit/261697/]
+
+ ```bash
+repository.findById
+
+or
+
+repository.findBy<ColumnNameId>
+
+//is internally a QueryDocument call and not a readDocument call
+
+```
+
+There is no option to use readDocument with partition key via spring-boot Java SDK spring-data-cosmosdb
+
+
+#### support for usage metrics like GetRequestCharge​ [ https://csefy19.visualstudio.com/CSE%20Feedback/_workitems/edit/261698/]
+
+Need support for Usage metrics like GetRequestCharge on the findBy calls, since findBy does not return ResourceResponse<Document> or ChangeFeed and returns of Type<T>
+
+
+#### No native query support with @Query Annotation [https://csefy19.visualstudio.com/CSE%20Feedback/_workitems/edit/254390/]
+
+ ```bash
+@Query(“Select * from c where c.userId= :1 order by order by c.createdAt ”)
+
+List<User> retrieveByID(string userId);
+
+```
+
+#### Lack of significant features , got the aggregation feature working now.
+Other limitations
+unique keys, no support for multiple partition keys, only strings supported in partition keys etc
+
+#### Similar lack of features have been observed on B2C, App Insights, App Config modules, Java Search SDKs
```

Comment:
Similar here - we should consider removing the citing of other projects in these docs, as well as the impact of Java.  It's definitely important we call out the impact/community points below in the meeting, but doesn't quite fit in a developer document.